### PR TITLE
Feat/relateditems

### PR DESCRIPTION
**Pull request by ryand8008**
Added ratings to each card, function current outfit to outfit list though it does not persist yet. Fixed buttons for comparison and removing item. Removing item is not 100% functional as it just resets the state and does not remove the individual item.

### CODE REVIEW

File: client/src/components/RelatedProducts/Cards/AllCards.jsx [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-  //     zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url]);
-  //   });
-  // }
-
-  if (thumbnail.data && relatedProds.data) {
+
+  if (thumbnail.data && relatedProds.data && ratings) {
     relatedProds.data.map((item, i) => {
-      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url]);
+      zippedArray.push([item.data, thumbnail.data[i].data.results[0].photos[0].thumbnail_url, ratings.data[i].data]);
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/AllCards.jsx:18]: The second inout seems to be a long one, recommend break this down a bit (save part to an easy to understand const name) to make the code easier to understand. Also seems a bit strange to be pulling from first item for both result and photo in a map.

File: client/src/components/RelatedProducts/Cards/AllCards.jsx [hunk 2]
```diff
@@ -39,6 +33,7 @@ function AllCards() {
 
 const Layout = styled.div`
   display: flex;
+  positive: relative;
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/AllCards.jsx:36]: Do you mean position?

File: client/src/components/RelatedProducts/Cards/Comparison.jsx [hunk 3]
```diff
@@ -1,33 +1,48 @@
...
+              <td>{item1Array.map((item1) => (item1.feature === item.feature ? 'X' : ' '))}</td>
 
-const Testing = styled.div`
-  border-style: solid;
-  height: max-content;
-  width: max-content;
-`;
+              <td>{item.value ? `${item.feature}: ${item.value}` : `${item.feature}`}</td>
+
+              <td>{item2Array.map((item2) => (item2.feature === item.feature ? 'X' : ' '))}</td>
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/Comparison.jsx:38]: It doesn't really matter it's limited to 2 products. I'd generally recommend to try to make the code DRY (i.e., 38 and 34 seems very similar, could potentially make number of products as variable to make it flexible say business requirement changed to four or five ).

File: client/src/components/RelatedProducts/Cards/FeatureContext.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+export function FeatureProvider({ children, prodID }) {
+  const [featuresInfo, setFeatures] = useState();
+
+  const productId = useContext(ProductIDContext);
+  const twoProducts = [productId, prodID];
+
+  useEffect(() => {
+    if (prodID) {
+      Promise.all(twoProducts.map((number) => axios({
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/FeatureContext.jsx:22]: Love this

File: client/src/components/RelatedProducts/Cards/Individualcard.jsx [hunk 5]
```diff
@@ -20,15 +24,19 @@ function Individualcard({ product }) {
             $
             {product[0].default_price}
           </p>
-          <p>Stars</p>
+          <div>
+            <Star score={starRating} />
+            {' '}
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/Individualcard.jsx:29]: Is this space intended?

File: client/src/components/RelatedProducts/Cards/ProdContext.jsx [hunk 6]
```diff
@@ -28,8 +30,47 @@ export function ProdProvider({ children }) {
...
+    let avgRating = 0;
+    let totalCT = null;
+    const { ratings } = data;
+    Object.keys(ratings).forEach((key) => {
+      const currentCT = Number(ratings[key]);
+      totalRating += key * currentCT;
+      totalCT += currentCT;
+    });
+    avgRating = Math.round((totalRating / totalCT) * 10) / 10;
+    return { avgRating, totalCT };
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/ProdContext.jsx:44]: I actually calculated avg-rating in ReviewMeta and pass it in as a property.  All components have access to it with RatingProvider. See use case in ratingList line 11-13

File: client/src/components/RelatedProducts/Cards/ProductImg.jsx [hunk 7]
```diff
@@ -2,51 +2,62 @@
...
   background-color: transparent;
   border: transparent;
+  margin: 5px;
 `;
 
-const TestingModal = styled.section`
-  display: ${(props) => (props.show ? 'block' : 'none')};
-  position: relative;
-  background: grey;
+const TestingModal = styled.div`
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Cards/ProductImg.jsx:53]: I can't seem to click anything - all click goes to google.com

File: client/src/components/RelatedProducts/Outfits/Outfit.jsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  const productToAdd = useCurrentProduct().currentProduct;
+  const initialArray = [];
+  const [outfitArray, setOutfitArray] = useState(initialArray);
+
+  return (
+
+    <AddSection>
+      <AddButton>
+
+        <img
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Outfits/Outfit.jsx:21]: Is it possible to refine the thumbnail display so its ratio is not distorted?

File: client/src/components/RelatedProducts/Outfits/OutfitCard.jsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+import { React } from 'react';
+import styled from 'styled-components';
+import { useCurrentStyles } from '../../../contexts/StylesProvider';
+import { useMeta } from '../../../contexts/ReviewMeta';
+import Star from '../../../Star';
+
+function OutfitCard({ product, setOutfitArray, outfitArray }) {
+  const thumbnail = useCurrentStyles()[0].photos[0].thumbnail_url;
+
+  const rating = useMeta().avgRating;
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Outfits/OutfitCard.jsx:13]: Oh nice! this is the avgRating I was referring to earlier

File: client/src/components/RelatedProducts/Outfits/OutfitCard.jsx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+  // use if add outfit array is more than 1
+  const removeItem = (product) => {
+    const remove = outfitArray.indexOf(product);
+    console.log(remove, 'this is 1');
```
Comment (ivyhu630) [inline client/src/components/RelatedProducts/Outfits/OutfitCard.jsx:18]: Please try to comment out all console log before pushing if possible

**Review by ivyhu630 — APPROVED** [review]
It's looking great! I can't seem to try out the compare function - when I click the card, it takes me to google. Could you take a look? Code looks super nice. I love how you are using Context consistently throughout.

File: client/src/components/RelatedProducts/Cards/AllCards.jsx (hunk 1 above)
Comment (ryand8008) [reply client/src/components/RelatedProducts/Cards/AllCards.jsx:18]: The modal window is displayed when you click on the star, maybe I'll adjust the size of the star when I start the fine tuning of CSS

File: client/src/components/RelatedProducts/Cards/Individualcard.jsx (hunk 5 above)
Comment (ryand8008) [reply client/src/components/RelatedProducts/Cards/Individualcard.jsx:29]: Linter was throwing an error and I just did autofix and it created it.

File: client/src/components/RelatedProducts/Outfits/Outfit.jsx (hunk 8 above)
Comment (ryand8008) [reply client/src/components/RelatedProducts/Outfits/Outfit.jsx:21]: I'll edit the thumbnail photo CSS, so it's not distorted when I start doing the final CSS. Walter suggested 'object-fit: cover'

**Pull request merged**